### PR TITLE
Added server to the options

### PR DESCRIPTION
**Pull request by quonic**
Created a simple option to tie over till full support for private servers.

### CODE REVIEW

File: tasks/screeps.js [hunk 1]
```diff
@@ -42,7 +42,7 @@ module.exports = function (grunt) {
             });
 
             var req = https.request({
-                hostname: 'screeps.com',
+                hostname: options.server,
```
Comment (fuchsnj) [inline tasks/screeps.js]: this should probably be `options.server || 'screeps.com'` to default to the official server

**Comment by artch** [comment]
Even if we merge this, there is no built-in code uploading support in private servers, and we don't plan to add it. Why would you need this? With private servers, all your scripts are just local files, you can simple work with them with directly.

**Comment by OverlordQ** [comment]
Then what does [this](https://github.com/screeps/backend-local/blob/69767811cb117336ffb8e7af50b18f0d122ff739/lib/game/api/user.js#L81) bit of code do?

**Comment by tedivm** [comment]
I have to say I would like this as well. I don't want to work on my code "live", so it lives in a different folder than the steam client. When I want to update it I run a grunt command manually, and would like to continue that experience with the private servers.

**Comment by artch** [comment]
@OverlordQ It deploys to the `POST /code` endpoint, but it uses Steam authentication, there is no built-in password authentication on private servers.

@tedivm Why not to "deploy" to your Steam client folder by simply copying the files when they are ready?

**Comment by tedivm** [comment]
I actually have a grunt task to do exactly that, but it actually hangs the steam client. Then when I log into the client again it asks me to overwrite my local code (which I obviously don't do). If I had to guess this is related to the size of my codebase (33605 lines, 219 files).

Also, while there is no built in password authentication there is at least one mod out there that adds that feature back in. It's already being used for people who want to do things like add stats collection or use the stand alone console, since both of those require some sort of password.

**Comment by tedivm** [comment]
Can you please merge this? I literally can not update the private server code right now and would really like to.

Again I tried to do what you said and copy/pasted the files into the appropriate directy, and again the steam client ended up just hanging completely.

**Comment by tedivm** [comment]
Well, I resolved the issue on my end by switching to an rsync module for the grunt task (so it would only copy changed files, not all of them). I still think this would be a useful feature though.

**Comment by ross-p** [comment]
I added PR #11 which is a more fully featured version of this change and is confirmed to work with [screepsmod-auth](https://github.com/screepsmods/screepsmod-auth)

**Comment by artch** [comment]
Accepted #11, closing this.

**Pull request closed without merging**